src/components/ble/NavigationService.h

```diff
@@ -26,7 +26,7 @@
 #undef max
 #undef min
 
-//c7e60000-78fc-48fe-8e23-43b37a1942d0
+//c7e60000-78fc-48fe-8e23-433b3a1942d0
 #define NAVIGATION_SERVICE_UUID_BASE {0xd0, 0x42, 0x19, 0x3a, 0x3b, 0x43, 0x23, 0x8e, 0xfe, 0x48, 0xfc, 0x78, 0x00, 0x00, 0xe6, 0xc7}
```

Comment:
If music uses c7e5xxxx-:scissors:-433b3a1942d0, call control uses c7e5xxxx-:scissors:-433b3a1942d1, could you please use c7e5xxxx-:scissors:-433b3a1942d3? It sounds more consistent to me personally.

Reply:
I was actually thinking about suggesting to change call control to c7**e7**xxxx-scissors-433b3a1942**d0**. As long as we iterate the same byte I guess its fine, but to me it seems easier to see the differences in UUID's with this variation. Might just be because of the 0's after.

Reply:
It kind-of makes more sense to increment the base than some other part. But in the end it does not matter much, if it's agreed upon now and assuming we won't receive thousands of services... there's more room to increment the base.

Reply:
I assume we should also the bundle all the UUID definition's in a dedicated file? Or at least documentation with all services listed in the same place

Reply:
Yes, having this documented would be very useful. A new documentation page?

Reply:
So, what the conclusion, what is the preference for UUID?

Reply:
I'm also wondering what is the state of the art to choose a custom UUID for a ble service.

Reply:
Ok, I read this comment again. If I understand correctly, the idea is to use the same "root" for all custom UUIDs defined in InfiniTime? 

So, for now, we have the following UUIDs :

```
- Music       c7e50000-78fc-48fe-8e23-433b3a1942d0
- Navigation  c7e60000-78fc-48fe-8e23-433b3a1942d0
- Call        c7e50000-78fc-48fe-8e23-433b3a1942d1
```

Note that for now, only the Music service has been released and implemented into companion apps.
@piggz and @petterhs chose to increment the 4th digit, while @Avamander and #161 chose the last one.

I honestly have no preference. @piggz choice is more similar than the standard ble 16bits uuid which also changes digits on "the left" for each service, and keep the same base on "the right" : 0000**XXXX**-0000-1000-8000-00805f9b34fb.

So if there's no good reason to change @piggz 's strategy, I think we can keep it.

Reply:
For a bit more background, this is how huami have numbered their services https://github.com/piggz/harbour-amazfish/blob/master/daemon/src/devices/bipdevice.h

Reply:
So they increment the digits [4..7], right?
It looks like most implementation read the uuid from right to left (when written in the human readable form), keep the first 96bits unchanged and number their services by changing 0000XXXX-0000-0000-0000-000000000000.
Do you think we should also follow this guideline?

Reply:
We could, though we have already started using 0000XXXX for the characteristics following the service uuid, where huani have completely different characteristic uuids.  So, we could continue using XXXX0000 for the services, and 0000XXXX for the chars?

Reply:
I like this aswell. And start the services from 0001XXXX instead of c7e5XXX

Reply:
Note that the Music service is already released and implemented in 2 companion apps (Amazfish and Gadgetbrigde). 
If we change this UUID, these apps will have to implement the new uuid and maybe keep the old one for retro-compatibility... Not good.

So, if we want to increment digits [4..7], we should use c7e5XXXX.

Reply:
I think it's fine if we break this compatibility once and force users to upgrade.

Reply:
So the final decision is:
```
- Music       00000000-78fc-48fe-8e23-433b3a1942d0
- Navigation  00010000-78fc-48fe-8e23-433b3a1942d0
- Call        00020000-78fc-48fe-8e23-433b3a1942d0
```
?

Reply:
I cant speak for GB, but i dont mind changing Amazfish if it means getting this right.
Things to consider
-Using a common base, with a separation of services and characteristics
-Ensuring services can be expanded in the future
I think this can be achieved using the 000X000Y- method of naming, which would mean changing the music service.   The same could probably be achieved without changing the music service by using c7e[x]000[y] where y = 1 for services and 2+ for characteristics, and X is incremented for each new service.

But, as I say, im happy to change it to whatever @JF002 decides.

Reply:
I think we agree to use the following scheme : SSSSCCCC-78fc-48fe-8e23-433b3a1942d0, where SSSS is a service ID and CCCC is a characteristic ID (where 0001 is reserved for the service).
This is how the Music service is setup : 

```
- c7e50000-78fc-48fe-8e23-433b3a1942d0 is the base UUID
- c7e50001-78fc-48fe-8e23-433b3a1942d0 is the UUID of the service
- c7e50002-78fc-48fe-8e23-433b3a1942d0 and following are the characteristics UUIDs
```

I think we can apply this scheme to other custom services : 

```
- c7e60001-78fc-48fe-8e23-433b3a1942d0 : service
- c7e60002-78fc-48fe-8e23-433b3a1942d0 : characteristic 1
- c7e60003-78fc-48fe-8e23-433b3a1942d0 : characteristic 2
- ...
```

Nothing prevent us from using 00000001-78fc-48fe-8e23-433b3a1942d0 or abcd0001-78fc-48fe-8e23-433b3a1942d0 as UUID for a custom service, but we can also use **c7e** as a signature for PineTime services (and deadbeef-78fc-48fe-8e23-433b3a1942d0 as easter egg :D )

This scheme allows us to keep the music service as is, and add new services and characteristics easily.
It also allows to keep the UUID of the navigation service as it is now.

Ok for everyone?

Reply:
> This scheme allows us to keep the music service as is [...] Ok for everyone?

I don't think this is a problem to change and would look a tiny bit neater, but sure.

So with current specific services it'd be:

```
- Music       c7e50000-78fc-48fe-8e23-433b3a1942d0
- Navigation  c7e60000-78fc-48fe-8e23-433b3a1942d0
- Call        c7e70000-78fc-48fe-8e23-433b3a1942d0
```

Correct?

Reply:
Changing means that companion app have to release a new version and maybe handle retrocompatibility with older fw. I'm not sure it worth it.

In fact, Music service uses c7e50000-... as a root UUID, but this one is never used. Instead the service is c7e50001 and chars are c7e50002, c7e50003,...

Reply:
And that is what this service does too, c7e60000 as the base, then c7e60001 for the service, and the characteristics after that.

Reply:
> In fact, Music service uses c7e50000-... as a root UUID, but this one is never used. Instead the service is c7e50001 and chars are c7e50002, c7e50003,

Let's leave it weird, but not continue like that?

Reply:
> Changing means that companion app have to release a new version and maybe handle retrocompatibility with older fw. I'm not sure it worth it.

I think it's better rip that bandaid off while we can, especially if use in production is low so far. If we need to shift music service around a bit, that's fine as well.

Reply:
I took some time to think about it, and, although there's no functional need to change the current scheme, it may be the only opportunity we have to start from scratch and avoid future debate and discussion about why c7exxxxx and why services starts at 1 and not 0... 
Here is my quick'n'dirty RFC : https://gist.github.com/JF002/c37d156c88f6dae7846e5252a7728163

This new scheme implies a modification of the MusicService. Hence, we'll need to update Amazfish and Gadgetbridge to take this changes into account.